sdk/storage/AzureStorageBlob/Source/DataStructures/URLSessionTransferManager.swift

```diff
@@ -0,0 +1,389 @@
+// --------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+//
+// The MIT License (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the ""Software""), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import AzureCore
+import CoreData
+
+public final class URLSessionTransferManager: NSObject, TransferManager, URLSessionTaskDelegate {
+    // MARK: Type Aliase
+
+    public typealias TransferManagerType = URLSessionTransferManager
+
+    // MARK: Properties
+
+    public weak var delegate: TransferManagerDelegate?
+
+    public var logger: ClientLogger
+
+    internal lazy var session: URLSession = {
+        let config = URLSessionConfiguration.background(withIdentifier: "com.azuresdk.transfermanager")
```

Comment:
If this doesn't have to be a triple, we should probably use one that matches or includes the bundle identifier.